lib/que/worker_group.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require_relative "worker"
+
+module Que
+  class WorkerGroup
+    DEFAULT_STOP_TIMEOUT = 5 # seconds
+
+    def self.start(count, **kwargs)
+      Que.logger.info(msg: 'Starting workers', event: 'que.worker.start', worker_count: count)
+
+      workers = Array.new(count) { Worker.new(**kwargs) }
+      worker_threads = workers.map { |worker| Thread.new { worker.work_loop } }
+
+      new(workers, worker_threads)
+    end
+
+    def initialize(workers, worker_threads)
+      @workers = workers
+      @worker_threads = worker_threads
+    end
+
+    attr_reader :workers
```

Comment:
I reckon we can also get rid of this, and just have a `collect_metrics` method on `WorkerGroup`